<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainForm</class>
 <widget class="QMainWindow" name="MainForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>779</width>
    <height>465</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>JSON</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>3</number>
      </property>
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tabCommon">
       <attribute name="title">
        <string>Общие</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="lbNameKod">
              <property name="text">
               <string>&amp;Название:</string>
              </property>
              <property name="buddy">
               <cstring>tbNameCod</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="tbNameCod"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="lbTimpCod">
              <property name="text">
               <string>Тайп-&amp;код:</string>
              </property>
              <property name="buddy">
               <cstring>tbTimpCod</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="tbTimpCod"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="lbGrop">
              <property name="text">
               <string>&amp;Группа:</string>
              </property>
              <property name="buddy">
               <cstring>tbGroup</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="tbGroup"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="lbVersion">
              <property name="text">
               <string>&amp;Версия:</string>
              </property>
              <property name="buddy">
               <cstring>tbVersion</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="tbVersion"/>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTableWidget" name="lwImport">
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <row>
             <property name="text">
              <string>1</string>
             </property>
            </row>
            <column>
             <property name="text">
              <string>Название файла для включения</string>
             </property>
            </column>
            <item row="0" column="0">
             <property name="text">
              <string>GenericHeader3</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QPushButton" name="btnImportitmAdd">
                <property name="text">
                 <string>&amp;Добавить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnImportitmDel">
                <property name="text">
                 <string>&amp;Удалить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnImportitmClean">
                <property name="text">
                 <string>&amp;Очистить</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabDicript">
       <attribute name="title">
        <string>Описание</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="lbNamePol">
            <property name="text">
             <string>&amp;Название поля:</string>
            </property>
            <property name="buddy">
             <cstring>tbNamePol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tbNamePol"/>
          </item>
          <item>
           <widget class="QLabel" name="lbTypePol">
            <property name="text">
             <string>&amp;Тип поля:</string>
            </property>
            <property name="buddy">
             <cstring>cbTypePol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbTypePol">
            <item>
             <property name="text">
              <string/>
             </property>
            </item>
            <item>
             <property name="text">
              <string>bool</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uint8_t</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uint16_t</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uint32_t</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="lbUnitPol">
            <property name="text">
             <string>&amp;Единица измерения</string>
            </property>
            <property name="buddy">
             <cstring>tbUnitPol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tbUnitPol"/>
          </item>
          <item>
           <widget class="QLabel" name="lbSizePol">
            <property name="text">
             <string>&amp;Размер поля:</string>
            </property>
            <property name="buddy">
             <cstring>spbSizePol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spbSizePol">
            <property name="suffix">
             <string> bit</string>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbLsbPol">
            <property name="text">
             <string>&amp;LSB:</string>
            </property>
            <property name="buddy">
             <cstring>tbLsbPol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tbLsbPol"/>
          </item>
         </layout>
        </item>
        <item row="2" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <item>
           <widget class="QLabel" name="lbCommentPol">
            <property name="text">
             <string>&amp;Коментарий</string>
            </property>
            <property name="buddy">
             <cstring>tbCommentPol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tbCommentPol"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnitmSave">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QTableWidget" name="twType">
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>80</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <row>
           <property name="text">
            <string>1</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string>Name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Unit</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Length</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>lsb</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>alignSize</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Comment</string>
           </property>
          </column>
          <item row="0" column="0">
           <property name="text">
            <string>Azimuth</string>
           </property>
          </item>
          <item row="0" column="1">
           <property name="text">
            <string>uint16_t</string>
           </property>
          </item>
          <item row="0" column="2">
           <property name="text">
            <string>Градусы</string>
           </property>
          </item>
          <item row="0" column="3">
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item row="0" column="4">
           <property name="text">
            <string>360.0/65536</string>
           </property>
          </item>
          <item row="0" column="6">
           <property name="text">
            <string>Азимут</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="btnitmAdd">
            <property name="text">
             <string>&amp;Добавить</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnitmDel">
            <property name="text">
             <string>&amp;Удалить</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnitmClean">
            <property name="text">
             <string>&amp;Очистить</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
       <zorder>twType</zorder>
       <zorder></zorder>
      </widget>
      <widget class="QWidget" name="tabEnum">
       <attribute name="title">
        <string>Перечисление</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QWidget" name="layoutWidget">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="lbListEnum">
              <property name="text">
               <string>Описание перечисления</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTableWidget" name="twListNum">
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>190</number>
              </attribute>
              <attribute name="horizontalHeaderMinimumSectionSize">
               <number>40</number>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Название перечисления</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Значение</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QPushButton" name="btnEnumitmAdd">
                <property name="text">
                 <string>&amp;Добавить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnEnumitmDel">
                <property name="text">
                 <string>&amp;Удалить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnEnumitmClean">
                <property name="text">
                 <string>&amp;Очистить</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="layoutWidget">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="lbListEnum_2">
              <property name="text">
               <string>Значение перечисления</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTableWidget" name="twDescpNum">
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>170</number>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Поле</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Значение</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QPushButton" name="btnEnumitmAdd_2">
                <property name="text">
                 <string>&amp;Добавить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnEnumitmDel_2">
                <property name="text">
                 <string>&amp;Удалить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnEnumitmClean_2">
                <property name="text">
                 <string>&amp;Очистить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnEnumitmSave_2">
                <property name="text">
                 <string>Save</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabType">
       <attribute name="title">
        <string>Подтипы</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QLabel" name="lbNameSubType">
            <property name="text">
             <string>&amp;Название типа:</string>
            </property>
            <property name="buddy">
             <cstring>tbNamePol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tbNameSubType"/>
          </item>
          <item>
           <widget class="QLabel" name="lbTypeSubType">
            <property name="text">
             <string>Базовый тип</string>
            </property>
            <property name="buddy">
             <cstring>cbTypePol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbTypeSubType">
            <item>
             <property name="text">
              <string/>
             </property>
            </item>
            <item>
             <property name="text">
              <string>bool</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uint8_t</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uint16_t</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>uint32_t</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <widget class="QLabel" name="lbSizeSubType">
            <property name="text">
             <string>&amp;Размер поля:</string>
            </property>
            <property name="buddy">
             <cstring>spbSizePol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spbSizeSubType">
            <property name="suffix">
             <string> bit</string>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbCommentSubType">
            <property name="text">
             <string>&amp;Коментарий</string>
            </property>
            <property name="buddy">
             <cstring>tbCommentPol</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tbCommentSubType"/>
          </item>
          <item>
           <widget class="QPushButton" name="btnSubTypeitmSave">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="2" column="0">
         <widget class="QSplitter" name="splitter_2">
          <property name="lineWidth">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QWidget" name="layoutWidget">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QTableWidget" name="twSubTypeList">
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>80</number>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Name subtype</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_17">
              <item>
               <widget class="QPushButton" name="btnSubTypeitmAdd_2">
                <property name="text">
                 <string>&amp;Добавить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnSubTypeitmDel_2">
                <property name="text">
                 <string>&amp;Удалить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnSubTypeitmClean_2">
                <property name="text">
                 <string>&amp;Очистить</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="layoutWidget">
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QTableWidget" name="twSubTypeTable">
              <property name="autoFillBackground">
               <bool>true</bool>
              </property>
              <attribute name="horizontalHeaderMinimumSectionSize">
               <number>80</number>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Type</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Length</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>alignSize</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Comment</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_15">
              <item>
               <widget class="QPushButton" name="btnSubTypeitmAdd">
                <property name="text">
                 <string>&amp;Добавить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnSubTypeitmDel">
                <property name="text">
                 <string>&amp;Удалить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnSubTypeitmClean">
                <property name="text">
                 <string>&amp;Очистить</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnSubTypeitmSave_2">
                <property name="text">
                 <string>Save</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>779</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Импорт</string>
    </property>
    <addaction name="actionInportENUM"/>
    <addaction name="actionInportType"/>
    <addaction name="actionInportDescript"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Экспорт</string>
    </property>
    <addaction name="actionExportJSON"/>
    <addaction name="actionExportDoc"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
  </widget>
  <action name="actionInportENUM">
   <property name="text">
    <string>ENUM</string>
   </property>
  </action>
  <action name="actionExportJSON">
   <property name="text">
    <string>JSON</string>
   </property>
  </action>
  <action name="actionExportDoc">
   <property name="text">
    <string>Doc</string>
   </property>
  </action>
  <action name="actionInportType">
   <property name="text">
    <string>SubType</string>
   </property>
  </action>
  <action name="actionInportDescript">
   <property name="text">
    <string>Descript</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainForm</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
